feat: check member permission funds periodically

0x-r4bbit opened the pull request:
This adds a periodic member permission check for every admin community such that member funds are checked every hour.

0x-r4bbit committed:
feat: check member permission funds periodically

This adds a periodic member permission check for every admin community
such that member funds are checked every hour.

ghost commented:
<!--prchecklist--> 
### Pull Request Checklist
- [ ] Have you updated the documentation, if impacted (e.g. [docs.status.im](https://status.im/docs/))?
- [ ] Have you tested changes with mobile?
- [ ] Have you tested changes with desktop?

status-im-auto commented:
### Jenkins Builds
<details>
<summary>Click to see older builds (14)</summary>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 94f0fde1 | [#1](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3317/1/) | 2023-03-20 12:30:42 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230320-122925-94f0fd-pr3317.zip) |
| :heavy_multiplication_x: | 94f0fde1 | [#1](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3317/1/) | 2023-03-20 12:32:08 | ~2 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3317/1//consoleText) |
| :heavy_check_mark: | 94f0fde1 | [#1](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3317/1/) | 2023-03-20 12:33:42 | ~4 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230320-122925-94f0fd-pr3317.aar) |
| :heavy_check_mark: | 94f0fde1 | [#1](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3317/1/) | 2023-03-20 12:33:43 | ~4 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230320-122925-94f0fd-pr3317.zip) |
| :heavy_multiplication_x: | 94f0fde1 | [#2](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3317/2/) | 2023-03-21 12:46:18 | ~22 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3317/2//consoleText) |
| :heavy_multiplication_x: | 94f0fde1 | [#3](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3317/3/) | 2023-03-22 09:43:03 | ~3 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3317/3//consoleText) |
| | | | | | | |
| :heavy_multiplication_x: | f9b2bc94 | [#4](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3317/4/) | 2023-03-22 12:01:42 | ~57 sec | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3317/4//consoleText) |
| :heavy_check_mark: | f9b2bc94 | [#2](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3317/2/) | 2023-03-22 12:01:59 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230322-120042-f9b2bc-pr3317.zip) |
| :heavy_check_mark: | f9b2bc94 | [#2](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3317/2/) | 2023-03-22 12:05:10 | ~4 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230322-120037-f9b2bc-pr3317.aar) |
| :heavy_check_mark: | f9b2bc94 | [#2](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3317/2/) | 2023-03-22 12:05:13 | ~4 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230322-120041-f9b2bc-pr3317.zip) |
| | | | | | | |
| :heavy_check_mark: | e30c622f | [#3](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3317/3/) | 2023-03-22 12:05:15 | ~57 sec | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230322-120417-e30c62-pr3317.zip) |
| :heavy_check_mark: | e30c622f | [#3](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3317/3/) | 2023-03-22 12:07:49 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230322-120515-e30c62-pr3317.zip) |
| :heavy_check_mark: | e30c622f | [#3](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3317/3/) | 2023-03-22 12:08:41 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230322-120515-e30c62-pr3317.aar) |
| :heavy_multiplication_x: | e30c622f | [#5](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3317/5/) | 2023-03-22 12:11:19 | ~7 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3317/5//consoleText) |
</details>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 093a028e | [#4](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3317/4/) | 2023-03-22 15:09:15 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230322-150813-093a02-pr3317.zip) |
| :heavy_check_mark: | 093a028e | [#4](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3317/4/) | 2023-03-22 15:11:36 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230322-150809-093a02-pr3317.zip) |
| :heavy_check_mark: | 093a028e | [#4](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3317/4/) | 2023-03-22 15:12:32 | ~4 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230322-150809-093a02-pr3317.aar) |
| :heavy_check_mark: | 093a028e | [#6](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3317/6/) | 2023-03-22 15:19:59 | ~11 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3317/6//consoleText) |
| :heavy_check_mark: | 093a028e | [#5](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3317/5/) | 2023-03-23 07:55:03 | ~59 sec | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230323-075404-093a02-pr3317.zip) |
| :heavy_check_mark: | 093a028e | [#5](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3317/5/) | 2023-03-23 07:56:29 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230323-075359-093a02-pr3317.zip) |
| :heavy_check_mark: | 093a028e | [#5](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3317/5/) | 2023-03-23 07:57:29 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230323-075359-093a02-pr3317.aar) |
| :heavy_check_mark: | 093a028e | [#7](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3317/7/) | 2023-03-23 08:15:27 | ~21 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3317/7//consoleText) |
| | | | | | | |
| :heavy_check_mark: | aeb39ac2 | [#6](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3317/6/) | 2023-03-23 07:56:11 | ~51 sec | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230323-075519-aeb39a-pr3317.zip) |
| :heavy_check_mark: | aeb39ac2 | [#6](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3317/6/) | 2023-03-23 07:59:01 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230323-075632-aeb39a-pr3317.zip) |
| :heavy_check_mark: | aeb39ac2 | [#6](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3317/6/) | 2023-03-23 08:00:35 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230323-075732-aeb39a-pr3317.aar) |
| :heavy_multiplication_x: | aeb39ac2 | [#8](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3317/8/) | 2023-03-23 08:37:20 | ~21 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3317/8//consoleText) |
| :heavy_check_mark: | aeb39ac2 | [#9](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3317/9/) | 2023-03-23 10:09:07 | ~2 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3317/9//consoleText) |

0x-r4bbit commented on protocol/communities/manager.go:
```diff
@@ -47,27 +47,30 @@ var pieceLength = 100 * 1024
 
 const maxArchiveSizeInBytes = 30000000
 
+var memberPermissionsCheckInterval = 1 * time.Hour
```
Wonder if we should go with a different interval here

Samyoul replied:
When implementing these kind of things you just pull a figure out of the air, it can be any reasonable number.

In this case, the functionality (If I'm reading it correctly) checks if a member's list contains members that have permission to be members, if not those members are removed.

So you are basically you are running a purge hourly. Is it possible to purge anytime the membership list is updated? Is that possible?

0x-r4bbit replied:
Thank you for sharing your thoughts on this Sam.
So the check is already happening when the membership list is updated (on handle request to join).

However, because the state of any member's wallet can change at any given time really, we have to regularly check on-chain if the all the required funds are still in each member's wallet.

In other words, with this we'd query various chains every hour.

Samyoul replied:
Ah, that is a very fair point. I was assuming permissions were soul bound but they may not be.

0x-r4bbit replied:
Humans don't have souls.

0x-r4bbit committed:
Do not automatically decline requests in manual accept

When community owners accept pending requests manually, they would be
declined in that process if the request doesn't fullfill the required
token permission criteria.

We don't want this to automatically reject those requests anymore,
instead, owners have to manually reject the requests.
